### PR TITLE
Fix isAdminUser.

### PR DESCRIPTION
**Pull request by neunhoef**
This attacks the problem that if a cluster is in read-only mode, no
user (except the superuser) had RW access to the system db. Therefore,
isAdminUser was returning false.

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9139/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9141/

File: arangod/RestHandler/RestReplicationHandler.cpp [hunk 1]
```diff
@@ -1049,7 +1049,7 @@ Result RestReplicationHandler::processRestoreCollection(VPackSlice const& collec
     return Result();
   }
 
-  ExecContextSuperuserScope escope(ExecContext::current().isAdminUser());
+  ExecContextSuperuserScope escope(ExecContext::current().isAdminUser() || !ServerState::readOnly());
```
Comment (jsteemann) [inline arangod/RestHandler/RestReplicationHandler.cpp]: Should this this really be an `||`? I would have expected `&&`.

File: arangod/RestServer/VocbaseContext.h [hunk 2]
```diff
@@ -45,7 +45,7 @@ class VocbaseContext : public arangodb::ExecContext {
   /// @brief upgrade to internal superuser
   void forceSuperuser();
 
-  /// @brief upgrade to internal read-only user
+  /// @brief upgrade to internal read-only useg
```
Comment (jsteemann) [inline arangod/RestServer/VocbaseContext.h]:
```suggestion
  /// @brief upgrade to internal read-only user
```

File: arangod/RestHandler/RestReplicationHandler.cpp (hunk 1 above)
Comment (neunhoef) [reply arangod/RestHandler/RestReplicationHandler.cpp]: Richtig. Ich fixe...

Comment (neunhoef) [reply arangod/RestHandler/RestReplicationHandler.cpp]: LOL, I have done this correctly in 3 places but in this one I made a boolean mistake.
Thanks for catching this!

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9144/

**Comment by neunhoef** [comment]
Still need to run `arangosync` tests on this.

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9148/

**Comment by neunhoef** [comment]
`arangosync` tests have been green for me with this version. This is good to go from my perspective.

**Pull request merged**